service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageController.kt

```diff
@@ -0,0 +1,153 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.messaging.message
+
+import fi.espoo.evaka.shared.Paged
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.auth.UserRole
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.Forbidden
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.util.UUID
+
+data class UnreadResponse(val unreadCount: Int)
+
+@RestController
+@RequestMapping("/messages")
+class MessageController {
```

Comment:
None of these endpoints require any specific ACL at the moment. Should they?

This is almost certainly a non-issue, but it seems that not all roles are relevant here. It might not be a terrible idea to limit these endpoints to the relevant roles.

Reply:
Good point. Created a separate task for authorization.